src/Previewer/vala-previewer/previewer.vala

```diff
@@ -3,56 +3,73 @@ namespace Workbench {
   [DBus (name="re.sonny.Workbench.vala_previewer")]
   public class Previewer : Object {
     construct {
-      this.window = new Gtk.Window () {
+      this.notify["ColorScheme"].connect (() => {
+        this.style_manager.color_scheme = this.ColorScheme;
+      });
+    }
+
+    private void ensure_window() {
+      if (this.window != null) {
+        return;
+      }
+      var window = new Gtk.Window () {
         // Ensure the header bar has the same height as Workbench
         titlebar = new Gtk.HeaderBar (),
         title = "Preview",
-        hide_on_close = true,
```

Comment:
@lw64 can you review/test the Vala / external preview side of this PR?

Reply:
It looks all right to me. Only a bit of moving code around into some methods. Except the `ensure_window()`, which is now called on demand instead of on creation of the process.

I see you have your own code style, so going with a prettyfier over it to establish it in the whole file would be good, right? (I used another style in the beginning and it is now a bit inconsistent.)